rust/src/database/verkle/variants/managed/nodes/leaf.rs

```diff
@@ -65,27 +68,27 @@ impl ManagedTrieNode for FullLeafNode {
         }
     }
 
-    fn next_store_action(
+    fn next_store_action<'a>(
         &self,
-        key: &Key,
+        updates: KeyedUpdateBatch<'a>,
         depth: u8,
         self_id: Self::Id,
-    ) -> BTResult<StoreAction<Self::Id, Self::Union>, Error> {
+    ) -> BTResult<StoreAction<'a, Self::Id, Self::Union>, Error> {
         // If key does not match the stem, we have to introduce a new inner node.
-        if key[..31] != self.stem[..] {
+        if !updates.all_stems_match(&self.stem) {
             let index = self.stem[depth as usize];
             let inner = InnerNode::new_with_leaf(index, self_id, &self.commitment);
             return Ok(StoreAction::HandleReparent(VerkleNode::Inner(Box::new(
                 inner,
             ))));
         }
 
-        Ok(StoreAction::Store {
-            index: key[31] as usize,
-        })
+        // All updates fit into this sparse leaf.
+        Ok(StoreAction::Store(updates))
     }
 
-    fn store(&mut self, key: &Key, value: &Value) -> BTResult<Value, Error> {
+    fn store(&mut self, update: &KeyedUpdate) -> BTResult<Value, Error> {
```

Comment:
In the in-memory implementation, the leaf receives the full `KeyedUpdateBatch`, whereas here each update is a separate `store` call - is that deliberate?